src/storage-blob-preview/azext_storage_blob_preview/_validators.py

```diff
@@ -1059,3 +1075,199 @@ def validate_upload_blob(namespace):
         raise InvalidArgumentValueError("usage error: please only specify one of --file and --data to upload.")
     if not namespace.file_path and not namespace.data:
         raise InvalidArgumentValueError("usage error: please specify one of --file and --data to upload.")
+
+
+def _match_path(path, pattern):
+    from fnmatch import fnmatch
+    return fnmatch(path, pattern)
+
+
+def glob_files_locally(folder_path, pattern):
+    """glob files in local folder based on the given pattern"""
+
+    pattern = os.path.join(folder_path, pattern.lstrip('/')) if pattern else None
+
+    len_folder_path = len(folder_path) + 1
+    for root, _, files in os.walk(folder_path):
+        for f in files:
+            full_path = os.path.join(root, f)
+            if not pattern or _match_path(full_path, pattern):
+                yield (full_path, full_path[len_folder_path:])
+
+
+def process_blob_download_batch_parameters(cmd, namespace):
+    """Process the parameters for storage blob download command"""
+    # 1. quick check
+    if not os.path.exists(namespace.destination) or not os.path.isdir(namespace.destination):
+        raise ValueError('incorrect usage: destination must be an existing directory')
```

Comment:
Could we use the more specific error type `InvalidArgumentValueError` instead of `ValueError`?

Reply:
updated